### PR TITLE
[#3642] [studio-ui] Add rexgex support to wcm-root-folder path excludes

### PR DESCRIPTION
**Pull request by vestrada**
https://github.com/craftercms/craftercms/issues/3642 - [studio-ui] Add rexgex support to wcm-root-folder path excludes #3642

### CODE REVIEW

File: static-assets/components/cstudio-contextual-nav/wcm-site-dropdown-mods/wcm-root-folder.js [hunk 1]
```diff
@@ -184,24 +185,13 @@
...
-                instance.excludeCache[path] = [];
-              }
-              instance.excludeCache[path].push(path);
-            });
-            
+          if (config.params.excludes && (typeof (config.params.excludes) == "object")){
+            if(config.params.excludes.exclude){
+              this.creatingExcludeArray(config.params.excludes.exclude, instance.excludeCache);
+            }
+            if(config.params.excludes.exclude){
```
Comment (rart) [inline static-assets/components/cstudio-contextual-nav/wcm-site-dropdown-mods/wcm-root-folder.js]: same if as above. did you mean `.regex`

File: static-assets/components/cstudio-contextual-nav/wcm-site-dropdown-mods/wcm-root-folder.js [hunk 2]
```diff
@@ -409,6 +399,42 @@
...
+      /**
+       * is Exclude
+       */
+      isExclude: function (path, instance) {
+        var exclude = false,
+            regex;
+        if(instance.excludeCache[path]) {
+          exclude = true;
+        }
+        for (const excludeRegexCache in instance.excludeRegexCache) {
```
Comment (rart) [inline static-assets/components/cstudio-contextual-nav/wcm-site-dropdown-mods/wcm-root-folder.js:428]: Use foreach

File: static-assets/components/cstudio-contextual-nav/wcm-site-dropdown-mods/wcm-root-folder.js [hunk 3]
```diff
@@ -409,6 +399,42 @@
...
+            }
+            excludeCache[path].push(path);
+          });
+        }
+      },
+
+      /**
+       * is Exclude
+       */
+      isExclude: function (path, instance) {
```
Comment (rart) [inline static-assets/components/cstudio-contextual-nav/wcm-site-dropdown-mods/wcm-root-folder.js]: Is this used only for regex type? Use better/more specific name

**Review by rart — CHANGES_REQUESTED** [review]

File: static-assets/components/cstudio-contextual-nav/wcm-site-dropdown-mods/wcm-root-folder.js (hunk 2 above)
Comment (vestrada) [reply static-assets/components/cstudio-contextual-nav/wcm-site-dropdown-mods/wcm-root-folder.js:428]: This is for objects .

**Review by rart — APPROVED** [review]

**Pull request merged**